ghedesigner/feature_recognition.py

```diff
@@ -1,45 +1,26 @@
-import cv2
-import numpy as np
+from ghedesigner.shape import point_polygon_check
 
 
-def scale_coordinates(coordinates, scale):
-    new_coordinates = []
-    for x, y in coordinates:
-        x *= scale
-        y *= scale
-        new_coordinates.append((x, y))
-    return new_coordinates
-
-
-def remove_cutout(coordinates, boundary=None, remove_inside=True, keep_contour=True):
-    if boundary is None:
-        boundary = []
-
-    # cv2.pointPolygonTest only takes integers, so we scale by 10000 and then
-    # scale back to keep precision
-    scale = 10000.0
-    coordinates = scale_coordinates(coordinates, scale)
-    boundary = scale_coordinates(boundary, scale)
-
-    _boundary = np.array(boundary, dtype=np.uint64)
```

Comment:
No need to scale the points up and down because we can now handle floating point values. This scaled the points up by 10000 and casts the number to an unsigned int... so negative-valued coordinates likely would have broken this.